drivers/can/can_kvaser_pci.c

```diff
@@ -172,7 +169,10 @@ const struct can_driver_api can_kvaser_pci_driver_api = {
 					       can_kvaser_pci_read_reg, can_kvaser_pci_write_reg,  \
 					       CAN_KVASER_PCI_OCR, CAN_KVASER_PCI_CDR);            \
                                                                                                    \
-	static struct can_kvaser_pci_config can_kvaser_pci_data_##inst;                            \
+	DEVICE_PCIE_INST_DECLARE(inst);                                                            \
+	static struct can_kvaser_pci_data can_kvaser_pci_data_##inst = {                           \
```

Comment:
@henrikbrixandersen note that this is also fixing a bug. The struct type was wrong, i.e. is should always have been "data" and not "config". This dropped between the cracks since it gets assigned to a void pointer and then in the functions back to the expected type. I guess the structs were close enough in size that there was no noticeable breakage, but there was a ticking time bomb here..

Reply:
Oops, thanks.

Reply:
@jhedberg Care to submit that bug fix as a separate PR since this one is still in draft?

Reply:
@henrikbrixandersen done: #52170